auth0/resource_auth0_action.go

```diff
@@ -251,6 +259,61 @@ func deployAction(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	return diag.FromErr(d.Set("version_id", actionVersion.GetID()))
 }
 
+func preventErasingUnmanagedSecrets(d *schema.ResourceData, api *management.Management) diag.Diagnostics {
+	if !d.HasChange("secrets") {
+		return nil
+	}
+
+	preUpdateAction, err := api.Action.Read(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oldSecrets, newSecrets := d.GetChange("secrets")
```

Comment:
What does "old" and "new" mean in this context? Is it the union of current, added and removed? If we're just trying to calculate all secrets that are and were managed, it might not make sense to continue to separate by "new" and "old" beyond this point because that really won't impact the logic below. 

My suggestion, assuming my understanding is correct, is to consolidate these into a single slice to simplify the logic below.

Reply:
Sure that makes sense. 

- Why do we need to get the change and not just the actual current secrets?
- If we use `d.Get()` we'll just get the secrets that are currently in the config, that means if we remove 1 we won't fetch it, but if we don't and the secret is actually on the action, we'll think that we're trying to wipe something that shouldn't get wiped, when actually we do want to wipe it. Hope this makes sense. Lmk if you have questions.